pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/OwnershipCache.java

```diff
@@ -238,7 +239,32 @@ public OwnershipCache(PulsarService pulsar, NamespaceBundleFactory bundleFactory
         }
 
         // If we're not the owner, we need to check if anybody else is
-        return resolveOwnership(path).thenApply(optional -> optional.map(Map.Entry::getKey));
+        CompletableFuture<Optional<NamespaceEphemeralData>> ownerFuture = new CompletableFuture<>();
+        ownershipReadOnlyCache.getAsync(path).whenComplete((owner, ex) -> {
+            if (ex != null) {
+                ownerFuture.completeExceptionally(ex);
+            } else if (!owner.isPresent()) {
+                ownerFuture.complete(Optional.empty());
+            } else {
+                try {
+                    URI ownerURI = new URI(owner.get().getHttpUrl());
+                    String ownerServiceAddr = ownerURI.getHost() + ":" + ownerURI.getPort();
+                    if (pulsar.getLoadManager().get().getAvailableBrokers().contains(ownerServiceAddr)) {
+                        ownerFuture.complete(owner);
+                    } else {
+                        LOG.warn("Namespace {} owned by broker {} which is not active, remove the ownership",
+                                suName.toString(), owner);
+                        removeOwnership(suName);
```

Comment:
here we are in `getOwnerAsync` that is meant to be free from side effects.

I am not sure this is the right fix